src/main/java/org/apache/maven/plugins/install/InstallMojo.java

```diff
@@ -83,24 +83,30 @@
     @Component
     private ProjectInstaller installer;
 
+    private enum State
+    {
+        SKIPPED, INSTALLED, TO_BE_INSTALLED
+    }
+
     public void execute()
         throws MojoExecutionException, MojoFailureException
     {
         if ( skip )
         {
-            getPluginContext().put( INSTALL_PROCESSED_MARKER, Boolean.FALSE );
             getLog().info( "Skipping artifact installation" );
+            getPluginContext().put( INSTALL_PROCESSED_MARKER, State.SKIPPED.name() );
```

Comment:
If we have objects, why not retain the enum, why use name?

Reply:
The span across several classloaders, there is no single enum.